web/stats.php

```diff
@@ -80,6 +98,27 @@
         require_once "../phplib/questionnaire_report_${previous_year}_WMC.php";
         $last_year = $GLOBALS["questionnaire_report_{$previous_year}_WMC"];
     }
+    # the current MP may not be the MP from the year of the stats so loop back
+    # through the reps till we get one who is in the stats. We need to check the
+    # area is the same on the small chance that e.g when asking for the 2007 stats
+    # the MP elected in 2015 was MP for a different seat and so you get their stats but
+    # for the wrong seat - e.g asking for stats for Uxbridge and South Ruislip for
+    # 2007 would get you Boris Johnson's stats for Henley in 2007. This only works if
+    # the cons name matches with MaPit which is should :/
+    if ($area_representatives) {
+        foreach ( $area_representatives as $rep ) {
+            $rep_info = dadem_get_representative_info($rep);
```

Comment:
Perhaps more straightforward to set something else here, and then if it matches set rep_info and break? Then you don't need to empty it on each iteration.